flaky/flaky_pytest_plugin.py

```diff
@@ -76,16 +76,15 @@ def pytest_runtest_protocol(self, item, nextitem):
                 self.min_passes,
             )
         original_call_runtest_hook = self.runner.call_runtest_hook
-        self._call_infos = {}
+        self._call_infos[item] = {}
         should_rerun = True
         try:
             self.runner.call_runtest_hook = self.call_runtest_hook
             while should_rerun:
                 self.runner.pytest_runtest_protocol(item, nextitem)
-                call_info = self._call_infos.get(self._PYTEST_WHEN_CALL, None)
-                if call_info is None:
+                run = self._call_infos.get(item, {}).get(self._PYTEST_WHEN_CALL, None)
```

Comment:
Why would `item` ever not exist in the dict? You just set it above, and you don't delete it from the dict until you exit this loop.